packages/x-date-pickers/src/AdapterDateFns/AdapterDateFns.ts

```diff
@@ -79,6 +80,14 @@ export class AdapterDateFns
   implements MuiPickersAdapter<Date, DateFnsLocale>
 {
   constructor({ locale, formats }: AdapterOptions<DateFnsLocale, never> = {}) {
+    if (!version.startsWith('2.')) {
```

Comment:
I guess we would bundle the whole `package.json` which is not ideal.
It would be great to be able to import the version from the package, but if they don't make this information available, maybe it's better to infer the major version from some API that changed rather than importing the `package.json`.

Reply:
Thanks for the tip.
I've used the imported function to identify the major: https://github.com/mui/mui-x/pull/11462/commits/22528bcfd6e8ba28bce24836ee8c021802c47c10
Seems to do the trick based on my testing. 💯 
WDYT? 🤔

Reply:
I think it's the easiest and best trick we can do :ok_hand: